docs/architecture/adr-005-subscription.md

```diff
@@ -0,0 +1,239 @@
+# ADR 5: Subscription
+
+## Changelog
+
+
+## Context
+
+Cosmos-SDK should allow developers to choose from a variety of payment models for their application. Subscriptions areo ne of the most popular payment models for Internet applications, so it is natural to provide this payment model as a module for sdk users.
+
+For fuller context around this issue: see [\#4642](https://github.com/cosmos/cosmos-sdk/issues/4642)
+
+## Decision
+
+Create Subscription module in SDK so that users can subscribe to various subscription services. Subscription collectors will send messages to collect subscription payments from users who have due subscriptions. Subscriptions are paid out directly from a user account. If the user does not have enough funds to pay for the subscription is inactivated. Users can define maximum limits on how many periods the subscription is valid for; if this limit elapses before a subscription renewal, the subscription is invalidated.
```

Comment:
Very important to give the power of cancelling subscriptions to the subscribing users not the receiver of subscription funds.. maybe worth adding this design consideration explicitly

Reply:
The only way a receiver of subscription funds can "cancel" a subscription is if the subscribing account has insufficient funds. This seems reasonable to me as the user can always refund the account and resubscribe.

The maximum limit is a user-defined parameter so that they can define an expiration date for their subscription. This is to mitigate the common problem of unused subscriptions siphoning off funds from account without user noticing.